lib/rules/property-units.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+var helpers = require('../helpers');
+
+module.exports = {
+  'name': 'property-units',
+  'defaults': {
+    'unitsAllowedForProperties': {},
+    'globallyAllowedUnits': []
+  },
+  'detect': function (ast, parser) {
+    var result = [];
+    var globallyAllowedUnits = parser.options.globallyAllowedUnits;
+    var unitsAllowedForProperties = parser.options.unitsAllowedForProperties;
+
+    ast.traverseByType('declaration', function (declaration) {
+      var declarationType = declaration.content[0].content[0].type,
+          propertyName = declaration.content[0].content[0].content,
```

Comment:
This is fairly dangerous, we should be checking each content block exists and is a type we are expecting. If any of these are undefined for whatever reason this will throw a fatal error. 

It's best here to use the ast's forEach method over the content blocks checking for the type you're looking for to make sure, at least that way we can fail gracefully and possibly have false positives rather than fatal errors.